sound/soc/sof/ipc4-topology.h

```diff
@@ -418,6 +418,27 @@ struct sof_ipc4_src {
 	struct sof_ipc4_msg msg;
 };
 
+/**
+ * struct sof_ipc4_asrc ASRC config data
+ * @base_config: IPC base config data
+ * @sink_rate: Output rate for sink module
+ * @asynchronous_mode: synchronous 0, asynchronous 1. When 1 the ASRC tracks and
+ * compensates for drift. When 0 can test ASRC with DAI endpoint without timestamps capability.
+ * @operation_mode: push 0, pull 1. In push mode the  ASRC consumes a defined number
+ * of frames at input, with varying number of frames at output. In pull mode the ASRC
+ * outputs a defined number of frames while number of input frames varies
+ * @available_fmt: Available audio format
+ * @msg: IPC4 message struct containing header and data info
+ */
+struct sof_ipc4_asrc {
+	struct sof_ipc4_base_module_cfg base_config;
+	uint32_t sink_rate;
+	uint32_t asynchronous_mode;
+	uint32_t operation_mode;
+	struct sof_ipc4_available_audio_format available_fmt;
+	struct sof_ipc4_msg msg;
+};
```

Comment:
I see, the first part of the message is used as a payload for the IPC message, I think this is how this should be represented:
```c
struct sof_ipc4_asrc_data {
	struct sof_ipc4_base_module_cfg base_config;
	uint32_t sink_rate;
	uint32_t asynchronous_mode;
	uint32_t operation_mode;
} __packed;

struct sof_ipc4_asrc {
	struct sof_ipc4_asrc_data data;
	struct sof_ipc4_available_audio_format available_fmt;
	struct sof_ipc4_msg msg;
};
```
Sidenote: Other modules have similar issues which needs to be addressed